warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java

```diff
@@ -170,39 +173,93 @@ public DateIndexHelper initialize(AccumuloClient client, String dateIndexTableNa
 
     public static class DateTypeDescription {
         final Set<String> fields = new HashSet<>();
-        final String[] dateRange = new String[2];
+        private String start;
+        private String end;
+
+        public void addField(String field) {
+            this.fields.add(field);
+        }
 
         public Set<String> getFields() {
             return fields;
         }
 
         public String[] getDateRange() {
-            return dateRange;
+            return new String[] {start, end};
+        }
+
+        /**
+         * Method used to set the default values for the start and end date if none were found
+         *
+         * @param startDate
+         *            the start date
+         * @param endDate
+         *            the end date
+         */
+        public void setStartEndDate(Date startDate, Date endDate) {
+            if (start == null) {
+                start = DateIndexUtil.format(startDate);
+            }
+            if (end == null) {
+                end = DateIndexUtil.format(endDate);
+            }
+        }
+
+        /**
+         * Method that updates the start and end date if the provided date is beyond either bound
+         *
+         * @param date
+         *            the date
+         */
+        public void updateStartEndDate(String date) {
```

Comment:
This method name is a bit ambiguous for what it does. Perhaps `ensureDateRangeIncludes(String date)`?